eclib/JArray.ec

```diff
@@ -523,3 +523,28 @@ abstract theory PolyArray.
 
 
 end PolyArray.
+
+(* Array slicing *)
+abstract theory SubArray.
+  op sizeS: int.
+  axiom gt0_sizeS: 0 < sizeS.
+
+  op sizeB: int.
+  axiom gt0_sizeB: 0 < sizeB.
+
+  (* Sub-array *)
+  clone import PolyArray as ArrayS with
+    op size <- sizeS
```

Comment:
I think op size <= sizeS will be better here. Because ArrayS will still be compatible with PolyArray.

Reply:
I don't have a good understanding of `<-` vs `<=` in theory cloning, but applying the suggested change leads to the following error:
```
[critical] [: line 1 (0) to line 7 (76)] In external theory SubArray2_3 [<unknown> location]:
unknown operator `Top.Array2.Array2.size'
```

in the following code (generated by jasmin2ec):
```
from Jasmin require import JArray.

require import Array2 Array3.

clone export SubArray as SubArray2_3  with op sizeS <- 2, op sizeB <- 3,
                                           theory ArrayS <= Array2,
                                           theory ArrayB <= Array3.
```

where the Array2 theory is defined as
```
from Jasmin require import JArray.

clone export PolyArray as Array2  with op size <- 2.
```

Maybe the problem comes from the way the `SubArray` or `Array` theories are cloned, I don't really understand how `<-` vs `<=` work. I found the current way to work after trial-and-error...

Besides what you can read in the patches for `JArray.ec` and `JWord_array.ec`, the code generated by `jasmin2ec` does the following (see [these functions](https://github.com/jasmin-lang/jasmin/blob/379462c2a854c613158c47dc31b1a06e6be4f6ba/compiler/src/toEC.ml#L714-L783)):

- `<-` for the array size & word size parameters (function `fmt_op`)
- `<-` for the "word" (W8/W16/...) theory instances (function `fmt_the`)
- `<=` for the other theories  (function `fmt_th`)

I'm open to any other solution (or explanation on what would/wouldn't work).